### PR TITLE
[TASK][26988] Signup

### PR DESCRIPTION
**Pull request by thuyptsun**
https://edu-redmine.sun-asterisk.vn/issues/26987
Screenshot from 2020-08-19 08-26-37
Screenshot from 2020-08-19 08-26-57
Screenshot from 2020-08-19 08-27-47

### CODE REVIEW

**Review by vinhcnt — APPROVED** [review]

**Comment by thuyptsun** [comment]
Ready

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -5,6 +5,7 @@ $fa-font-path: "@fortawesome/fontawesome-free/webfonts";
 @import "@fortawesome/fontawesome-free/scss/brands";
 
 @import "bootstrap/scss/bootstrap";
+@import "toastr/toastr.scss";
 
 @import "./style.scss";
```
Comment (saobv-2223) [inline app/assets/stylesheets/application.scss]: dùng '

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  def create
+    @user = User.new candidate_params
+    if @user.save
+      @user.send_activation_email
+      flash[:success] = t ".success"
+      redirect_to root_path
+    else
+      render :new
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb:14]: thêm flash nữa e

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -3,4 +3,14 @@ def full_title page_title
     base_title = t "title"
     page_title.blank? ? base_title : page_title + " | " + base_title
   end
+
+  def toastr_flash
+    flash_script = flash.each_with_object([]) do |(type, message), flash_messages|
+      type = "success" if type == "notice"
+      type = "error" if type == "alert"
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]: dùng .eql? đi e

File: app/models/user.rb [hunk 4]
```diff
@@ -1,3 +1,64 @@
...
+
+  validates :password, presence: true,
+    length: {minimum: Settings.validations.user.password_minlength,
+             maximum: Settings.validations.user.password_maxlength},
+    allow_nil: true
+
+  before_save :downcase_email
+  before_create :create_activation_digest
+
+  has_secure_password
```
Comment (saobv-2223) [inline app/models/user.rb]: sắp xếp lại cho đúng nhé

**Comment by thuyptsun** [comment]
Ready

**Pull request merged**